sql/sql_parse.cc

```diff
@@ -1272,14 +1272,14 @@ bool dispatch_command(enum enum_server_command command, THD *thd,
     {
       my_error(ER_LOCK_DEADLOCK, MYF(0), "wsrep aborted transaction");
       WSREP_DEBUG("Deadlock error for: %s", thd->query());
-      mysql_mutex_unlock(&thd->LOCK_wsrep_thd);
+      mysql_mutex_unlock(&thd->LOCK_thd_data);
```

Comment:
Here and in some other places, why do you need mutex protection for `my_error()` and `WSREP_DEBUG()`?

Specifically `my_error()` may lock many different mutexes/rwlocks (e.g. `LOCK_plugin` for auditing purposes, query cache locks, etc).